src/Microsoft.Sbom.Api/PackageDetails/ComponentDetailsUtils/RubyGemsUtils.cs

```diff
@@ -0,0 +1,291 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text.RegularExpressions;
+using Microsoft.ComponentDetection.Contracts.BcdeModels;
+using Microsoft.Sbom.Api.Exceptions;
+using Microsoft.Sbom.Api.Output.Telemetry;
+using Microsoft.Sbom.Common;
+using Serilog;
+using File = System.IO.File;
+
+namespace Microsoft.Sbom.Api.PackageDetails;
+
+/// <summary>
+/// Utilities for retrieving information from rubygems packages that may not be present on the buildDropPath.
+/// </summary>
+public class RubyGemsUtils : IPackageManagerUtils<RubyGemsUtils>
+{
+    private const int ExecutableIndex = 1;
+
+    private readonly IFileSystemUtils fileSystemUtils;
+    private readonly ILogger log;
+    private readonly IRecorder recorder;
+
+    private readonly string rubyGemsPath;
+    private string[] potentialGemEnvPaths;
+
+    public RubyGemsUtils(IFileSystemUtils fileSystemUtils, ILogger log, IRecorder recorder)
+    {
+        this.fileSystemUtils = fileSystemUtils ?? throw new ArgumentNullException(nameof(fileSystemUtils));
+        this.log = log ?? throw new ArgumentNullException(nameof(log));
+        this.recorder = recorder ?? throw new ArgumentNullException(nameof(recorder));
+
+        this.rubyGemsPath = GetRubyGemsSpecificationsPath();
```

Comment:
I would advise not calling methods that do `Process.Start` from the constructor. There's a lot of failure-cases and caveats for `Process.Start` and people generally treat a constructor more reliable than that. Maybe make a property for RubyGemsPath which calls GetRubyGemsSpecificationsPath if the underlying variable hasn't already been populated.